runtime-sdk/modules/evm/src/precompile/erc20.rs

```diff
@@ -0,0 +1,435 @@
+    approve: u64,
+    allowance: u64,
+    mint: u64,
+    burn: u64,
+}
+
+impl TokenOperationCosts {
+    pub const fn new() -> Self {
+        Self {
+            name: 0,
+            symbol: 0,
+            decimals: 0,
+            total_supply: 0,
+            balace_of: 0,
+            transfer: 0,
+            transfer_from: 0,
+            approve: 0,
+            allowance: 0,
+            mint: 0,
+            burn: 0,
+        }
+    }
+}
+
+/// Static ERC-20 token implementations should implement this trait to provide
+/// token-specific configuration to the static contract framework.
+pub trait Erc20Token {
+    const GAS_COSTS: TokenOperationCosts;
+
+    const NAME: &str;
+    const SYMBOL: &str;
+    const DECIMALS: u8;
+
+    fn address() -> H160;
+
+    fn total_supply() -> Result<u128, Error>;
+    fn balance_of(account: address::Address) -> Result<u128, Error>;
+    fn transfer(
+        sender: address::Address,
+        recipient: address::Address,
+        amount: &token::BaseUnits,
+    ) -> Result<bool, Error>;
+    fn approve(
+        owner: address::Address,
+        spender: address::Address,
+        amount: &token::BaseUnits,
+    ) -> Result<bool, Error>;
+    fn allowance(owner: address::Address, spender: address::Address) -> Result<u128, Error>;
+    fn mint(to: address::Address, amount: &token::BaseUnits) -> Result<(), Error>;
+    fn burn(from: address::Address, amount: &token::BaseUnits) -> Result<(), Error>;
+}
+
+pub trait AccountToken {
+    type Accounts: AccountsAPI;
+
+    const ADDRESS: H160;
+
+    const GAS_COSTS: TokenOperationCosts;
+
+    const NAME: &str;
+    const SYMBOL: &str;
+    const DECIMALS: u8;
+}
+
+impl<T: AccountToken> Erc20Token for T {
+    const GAS_COSTS: TokenOperationCosts = T::GAS_COSTS;
+
+    const NAME: &str = T::NAME;
+    const SYMBOL: &str = T::SYMBOL;
+    const DECIMALS: u8 = T::DECIMALS;
+
+    fn address() -> H160 {
+        Self::ADDRESS
+    }
+
+    fn total_supply() -> Result<u128, Error> {
+        let denom = token::Denomination::from_str(Self::SYMBOL).unwrap();
+        Ok(T::Accounts::get_total_supplies()?
+            .get(&denom)
+            .copied()
+            .unwrap_or_default())
+    }
+
+    fn balance_of(account: address::Address) -> Result<u128, Error> {
+        let denom = token::Denomination::from_str(Self::SYMBOL).unwrap();
+        T::Accounts::get_balance(account, denom)
+    }
+
+    fn transfer(
+        sender: address::Address,
+        recipient: address::Address,
+        amount: &token::BaseUnits,
+    ) -> Result<bool, Error> {
+        T::Accounts::transfer(sender, recipient, amount).map(|_| true)
+    }
+
+    fn approve(
+        _owner: address::Address,
+        _spender: address::Address,
+        _amount: &token::BaseUnits,
+    ) -> Result<bool, Error> {
+        Ok(false) // XXX
+    }
+
+    fn allowance(_owner: address::Address, _spender: address::Address) -> Result<u128, Error> {
+        Ok(0) // XXX
```

Comment:
Do we plan to support this?  Otherwise, it might it be better to fail on the unsupported methods?